libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/lifecycle/UIViewModel.kt

```diff
@@ -39,6 +41,25 @@ abstract class UIViewModel<State, Action>(initialState: State) : MapboxNavigatio
         _action.tryEmit(action)
     }
 
+    /**
+     * Actions that are invoked inside [onAttached] or [onDetached] callbacks can be dropped.
+     * There is also, no feedback from what the action did to the state when using [invoke].
+     */
+    fun invokeSync(action: Action): State {
```

Comment:
If this were the approach we took with this function, we actually would have never ran into the issues 🤔

Reply:
This is a potential approach, realized that all actions can be invoked immediately